File: Data Structures & Algorithms/check-if-array-is-sorted-and-rotated/submission-0.cs
public class Solution {
    public bool Check(int[] nums) {
        int n = nums.Length, count  = 0;        
        for(int i  = 1; i < 2 * n; i++){// 34512 nối tiếp 34512 
            if(nums[(i - 1) % n] <= nums[i % n]) count++; // check liên tiếp 
            else count = 1; // reset count to check sort 345 ok reset từ 12 đếm tiếm 12345 
            if(count == n) return true; // check = length return 
        }
        return n == 1; //if = 1 => true
    }
}

File: Data Structures & Algorithms/subarray-sum-equals-k/submission-0.java
class Solution {
    public int subarraySum(int[] nums, int k) {
        int count = 0, sum = 0;
        Map<Integer, Integer> prefixSum = new HashMap<Integer, Integer>();
        prefixSum.put(0, 1);//empty prefix
        for(int num : nums){
            sum += num;
            int diff = sum - k;
            count += prefixSum.getOrDefault(diff, 0);
            prefixSum.put(sum, prefixSum.getOrDefault(sum, 0) + 1);
        }
        return count;
    }
}